frameworks/hdfs/alerts/HDFS_alert_rules.yml

```diff
@@ -0,0 +1,96 @@
+       description: "MissingBlocks value has increased for the last 10 min, you can lose your data"
+
+    - alert: UnderReplicatedBlocks
+      expr: sum by(service_name,task_name) (rate({__name__=~".*_dfs_FSNamesystem_UnderReplicatedBlocks"}[10m])) > 0
+      labels:
+        severity: warning
+        service: hdfs
+      annotations:
+       summary: "Under replicated blocks on {{ $labels.instance }} for {{ $labels.service_name }} / {{ $labels.task_name }}"
+       description: "UnderReplicatedBlocks value has increased for the last 10 min, please consider to check the cluster state"
+
+    - alert: NumStaleDataNodes
+      expr: sum by(service_name,task_name) (rate({__name__=~".*_dfs_FSNamesystem_StaleDataNodes"}[10m])) > 0
+      labels:
+        severity: warning
+        service: hdfs
+      annotations:
+       summary: "Number of stale DataNodes on {{ $labels.instance }} for {{ $labels.service_name }} / {{ $labels.task_name }}"
+       description: "DataNode might not emit a heartbeat for a number of reasons — network issues or high load are among the more common causes"
+
+    - alert: MemHeapUsed
+      expr: (sum({__name__=~".*_jvm_JvmMetrics_MemHeapMaxM"}) / sum({__name__=~".*_jvm_JvmMetrics_MemHeapUsedM"})) < 0.8
+      for: 10m
+      labels:
+        severity: warning
+        service: hdfs
+      annotations:
+       summary: "Total service memory heap in use for {{ $labels.instance }}"
+       description: "Total memory heap usage is greater than 80%. Consider increasing memory of the HDFS cluster"
+
+    - alert: MemUsedPercent
+      expr: mem_used_percent > 80
+      for: 10m
+      labels:
+        severity: warning
+        service: hdfs
+      annotations:
+       summary: "Instance memory in use percent for {{ $labels.instance }}"
+       description: "Total memory heap usage is greater than 80%. Consider increasing memory of the HDFS cluster"
+
+  - name: hdfs_critical
+    rules:
+
+    - alert: NumInactiveSources
+      expr: (sum({__name__=~".*metricssystem_MetricsSystem_NumAllSources"}) - sum({__name__=~".*metricssystem_MetricsSystem_NumActiveSources"})) > 0
+      for: 5m
+      labels:
+        severity: critical
+        service: hdfs
+      annotations:
+       summary: "Number of inactive sources for {{ $labels.instance }}"
+       description: "NumInactiveSources value has increased for the last 5 min, please consider to check the cluster state"
+
+    - alert: VolumeFailures
+      expr: sum by (service_name,task_name) ({__name__=~".*_dfs_datanode_VolumeFailures"}) > 0
+      for: 5m
+      labels:
+        severity: critical
+        service: hdfs
+      annotations:
+       summary: "VolumeFailures on {{ $labels.instance }} for {{ $labels.service_name }} / {{ $labels.task_name }}"
+       description: "Hardware failures occur, please consider replace the failed hardware. Failed volume is not a data loss signal"
+
```

Comment:
Let's add one more critical alert for fatal logs eg
```
    - alert: FatalLogErrors
      expr: {__name__=~".*(DataNode|JournalNode|NamdeNode)*jvm_JvmMetrics_LogFatal"} > 0
      for: 1m
      labels:
        severity: critical
        service: hdfs
      annotations:
       summary: "Fatal log errors have been reported on {{ $labels.instance }} for {{ $labels.service_name }} / {{ $labels.task_name }}"
       description: "A fatal log message has been reported. Please investigate as this may indicate a serious issue"
```